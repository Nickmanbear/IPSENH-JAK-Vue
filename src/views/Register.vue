<template>
  <div id="register">
    <div class="container">
      <form>
        <h1>Signup</h1>
        <hr>
        <div>
          <label for="username">Username</label>
          <input type="text" id="username" v-model="userData.username">
        </div>
        <div>
          <label for="password">Password</label>
          <input type="password" id="password" v-model="userData.password">
        </div>
        <hr>
        <div class="row">
          <button @click.prevent="register">Submit!</button>
        </div>
      </form>
    </div>
  </div>
</template>

<script>

export default {
  name: 'Register',
  data() {
    return {
      userData: {
        username: '',
        password: '',
      },
    };
  },

  methods: {
    register() {
      this.$store.dispatch('register', {
        username: this.userData.username,
        password: this.userData.password,
      })
        .then(() => {
          this.$router.push('/login');
        });
    },
  },
};
</script>

<style>
  #register {
    margin: auto;
    width: 500px;
  }
</style>
