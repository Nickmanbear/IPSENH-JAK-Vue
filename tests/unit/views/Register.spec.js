import { shallowMount } from '@vue/test-utils';
import Register from '@/views/Register.vue';
import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

describe('Register', () => {
  let cmp;
  let store;
  let actions;
  let state;
  beforeEach(async () => {
    state = { data: {} };
    actions = {
      register: jest.fn(),
    };
    store = new Vuex.Store({
      state,
      actions,
    });
    cmp = await shallowMount(Register, {
      data() {
        return {
          userData: {
            username: 'username',
            password: 'password',
          },
        };
      },
      store,
    });
  });

  it('should Register a user ', async () => {
    cmp.find('button').trigger('click');
    expect(actions.register.mock.calls).toHaveLength(1);
  });
  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
