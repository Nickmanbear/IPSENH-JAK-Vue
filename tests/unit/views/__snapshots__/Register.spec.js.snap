// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register has the expected html structure 1`] = `
<div
  id="register"
>
  <div
    class="container"
  >
    <form>
      <h1>
        Register
      </h1>
       
      <input
        id="username"
        placeholder="Username"
        type="text"
      />
       
      <!---->
       
      <input
        id="password"
        placeholder="Password"
        type="password"
      />
       
      <button>
        Sign up!
      </button>
    </form>
  </div>
</div>
`;
